src/main/java/org/rumbledb/expressions/CommaExpression.java

```diff
@@ -64,9 +64,17 @@ private boolean bypassCurrentExpressionForExecutionModeOperations() {
     @Override
     public void initHighestExecutionMode() {
         if (bypassCurrentExpressionForExecutionModeOperations()) {
+            this.highestExecutionMode = this.expressions.get(0).getHighestExecutionMode();
             return;
         }
-        super.initHighestExecutionMode();
+
+        ExecutionMode currentMode = ExecutionMode.LOCAL;
```

Comment:
I think that, for a first implementation, it will be easier to do the opposite: only do RDD if all children expressions are RDDs, and otherwise local.

This is because the code can currently only materialize from RDD to local, but not the other way round.